api/Onboard.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/operate-first/opfcli/constants"
+	"github.com/operate-first/opfcli/models"
+	"github.com/operate-first/opfcli/utils"
+)
+
+// TODO: use kustomization package so we can unmarshal kustomizations properly, and not just resources or : https://pkg.go.dev/github.com/williepaul/kustomize/api/types
+
+func (api *API) Onboard(path string, projectDisplayName string, DisableLimitrange bool) error {
+	onboardRequest, err := models.OnboardRequestFromYAMLPath(path)
+
+	if err != nil {
+		return err
+	}
+
+	if err := api.CreateGroup(
+		onboardRequest.TeamName,
+		onboardRequest.Users,
+		true,
+	); err != nil {
+		return err
+	}
+
+	if err := api.CreateRoleBinding(onboardRequest.TeamName, "admin"); err != nil {
+		return err
+	}
+
+	var tempQuota string
+	var nsPaths []string
+	emptyCustomQuota := models.CustomResourceQuota{}
+
+	for i := range onboardRequest.Namespaces {
+		if onboardRequest.Namespaces[i].CustomQuota != emptyCustomQuota {
+			tempQuota = ""
+		} else if onboardRequest.Namespaces[i].Quota != "" && onboardRequest.Namespaces[i].Quota != "_No response_" {
+			tempQuota = strings.ToLower(onboardRequest.Namespaces[i].Quota)
+			if err := api.ValidateQuota(tempQuota); err != nil {
+				return err
+			}
+		}
+		if err := api.CreateNamespace(
+			onboardRequest.Namespaces[i].Name,
+			onboardRequest.TeamName,
+			projectDisplayName,
+			tempQuota,
+			DisableLimitrange,
+			true,
+		); err != nil {
+			return err
+		}
+		if onboardRequest.Namespaces[i].CustomQuota != emptyCustomQuota {
```

Comment:
seems like you are checking for `emptyCustomQuota` twice, see line 39

Reply:
I know its not the cleanest implementation but this is actually intentional. This is because `api.CreateNamespace` is checking for pre-configured resource quotas, such as `x-small`, `small`, `medium`,  and `large` defined [here](https://github.com/operate-first/apps/tree/master/cluster-scope/components/resourcequotas) in `operate-first/apps`. However custom quotas live in the namespace directory, rather than the `cluster-scope/components/resourcequotas` directory. This is why if there is a custom quota, we set `tempQuota` to an empty string, to signify that this is not "operator defined" resource quota, but instead a custom one. The changes you have referenced here (line 57-59) are what actually creates the custom quota resource and appends it to the `kustomization.yaml` available in the namespace.

Reply:
okay then I suggest cleaning it up a bit further, maybe introduce a variable, something like: 


```go
		var customQuotaRequested = onboardRequest.Namespaces[i].CustomQuota != emptyCustomQuota
		tempQuota = ""
		if !customQuotaRequested && onboardRequest.Namespaces[i].Quota != "" {
			tempQuota = strings.ToLower(onboardRequest.Namespaces[i].Quota)
			if err := api.ValidateQuota(tempQuota); err != nil {
				return err
			}
		}

		if err := api.CreateNamespace(
			onboardRequest.Namespaces[i].Name,
			onboardRequest.TeamName,
			projectDisplayName,
			tempQuota,
			DisableLimitrange,
			true,
		); err != nil {
			return err
		}

		if customQuotaRequested {
			if err := api.CreateCustomResourceQuota(onboardRequest.Namespaces[i].Name, onboardRequest.Namespaces[i].CustomQuota, true); err != nil {
				return err
			}
		}
```

(note: I didn't test this, but you get the idea)